ADBench/run-all.ps1

```diff
@@ -189,12 +189,7 @@ Class Tool {
 		$cmd = ""
 		$cmdargs = @($dir_in, $dir_out, $fn, $script:nruns_f, $script:nruns_J, $script:time_limit)
 		if ($this.type -eq "bin") {
-			if ($this.name -eq "DiffSharp") {
-				$builddir = ""
-			} else {
-				$builddir = "\$script:buildtype"
-			}
-			$cmd = "$script:bindir\tools\$($this.name)$builddir\Tools-$($this.name)-$objective.exe"
+			$cmd = "$script:bindir/tools/$($this.name)/Tools-$($this.name)-$objective.exe"
```

Comment:
Great to get rid of this horror, but would expect to see a change to a fsproj somewhere ensuring the diffsharp exe was put in the right place.

Reply:
This change is behaviour-preserving for the diffsharp branch.  It only changes behaviour when `$this.name` is *not* `"DiffSharp"`.

Reply:
I see, but now it's quite wide-ranging.  The buildtype argument now does very little.  To make a "Debug" graph (not that one really wants one, but it was an option before), one would now need to run cmake in a new build directory, with new default build arguments.

That would really imply that the artifacts should be output in the build directory, not the top-level tmp dir, which itself makes good sense.

Reply:
For me the binaries appear in places like `tools/Manual/Tools-Manual-GMM.exe`.  Apparently this is not the case for all versions of VS. @msdmkats has a PR (https://github.com/awf/ADBench/pull/39) to make it the case for all versions.   Would you prefer Release and Debug builds to go in separate subdirectories, i.e.

* `tools/Manual/Release/Tools-Manual-GMM.exe`
* `tools/Manual/Debug/Tools-Manual-GMM.exe`

If so then we will probably need to fix that PR.

Reply:
It was not a matter of the version of VS, but the cmake's generator used. Some generators, namely Visual Studio/msbuild are multi-configuration generators, i.e. they allow to do builds in different configurations with only one run of cmake, However, most generators (Ninja - the default one, Unix Make) don't have such functionality. And they require separate runs of cmake anyway.

Reply:
Thanks for the clarification.  I think it's now up to @awf to decide how to proceed.  If we're trying to support Linux I don't think we'll be able to support this distinction between Debug and Release.

Reply:
Yes, so I propose to remove "buildtype" from run-all, and make this change.
That will have a further implication that run-all will probably need to be run from inside a cmake build dir, so let's put the "cmake-vars-BUILD.ps1" script into that build dir rather than in the ADbench dir, and then include "./cmake-vars.ps1" in run-all.  If it doesn't exist, then have run-all emit an error saying "This script should be run inside a cmake build directory".

Reply:
I know it's a maze of twisty passages, but I believe we're removing some of the twists.

Reply:
OK, I think that can be part of @msdmkats's ongoing work.

Reply:
I cancelled the removal of the build type dependent code from this PR.  We can deal with it elsewhere.

Reply:
@awf I think we're good to go on this one.  Can you confirm?